gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapper.java

```diff
@@ -190,6 +190,13 @@ private DateTimeType getAssertedDateElement(TS availabilityTime, RCMRMT030101UK0
     private void buildAllergyIntoleranceText(RCMRMT030101UK04ObservationStatement observationStatement,
```

Comment:
Do the tests pass if we remove this method entirely? I'm confused as to why we need to manipulate the code text at all.

Reply:
It looks like the codeable concept mapper does not map the text field as is mapped using a conceptId and no original text has been entered.  In this case the text value is set to null.

I've tried multiple ways of changing this in the codeable concept mapper, but results in a lot of changes to the output for other elements which use codeable concept mapper.

For allergyIntolerance, this issue requires this text value to be set, so unfortunately I believe we need to leave this in.

I have refactored the buildCode method in the allergyIntolerance mapper to make it clear to understand and remove some nesting